### ISSUE TITLE
Tigera operator Helm uninstall does not complete cleanly for v3.27.3

### ISSUE THREAD

**Issue by pfrydids** [projectcalico/calico#8695]
## Expected Behavior
Uninstalling the Tigera operator via helm should work

## Current Behavior
Does not complete and leaves

```
calico-system       calico-typha-69cfd4c97-jgj5x                         1/1     Running   0          46m
calico-system       calico-typha-69cfd4c97-qk7jl                         1/1     Running   0          48m
```

## Your Environment
* EKS 1.26
* Calico version v3.27.3

This is related to 

https://github.com/projectcalico/calico/pull/8413 which targets 3.28.x

and

https://github.com/projectcalico/calico/pull/8547 which was included in 3.27 and I was hoping was going to fix the uninstallation issues.

**Comment by caseydavenport** [comment]
I mentioned in my comment on that issue that there were actually other issues preventing uninstall as well.

Namely, a problem with finalizers due to the Calico API server that was fixed in this PR: https://github.com/projectcalico/calico/pull/8586

That fix will be in v3.28 - please give this a re-test then.

In the meantime, you may need to manually remove finalizers from and delete the following resources before the calico-system resources will terminate:
- `calico-apiserver` namesapce
- ServiceAccounts within the `calico-apiserver` namespace

**Comment by caseydavenport** [comment]
Closing this for now as it will be resolved in v3.28 via: https://github.com/projectcalico/calico/pull/8586

That PR may be cherry-picked back to v3.27 but it needs some more soak time first.
